tests/runner/test_dataset_block.py

```diff
@@ -103,6 +127,113 @@ def test_partial_block_for_chunked_data(slicing_dim: Literal[0, 1, 2], last_in_c
+        chunk_shape=chunk_shape,
+        padding=(2, 2),
+    )
+
+    assert block.is_padded is True
+    assert block.padding == (2, 2)
+
+
+@pytest.mark.parametrize("slicing_dim", [0, 1, 2])
+@pytest.mark.parametrize("boundary", ["before", "after"])
+def test_partial_block_for_chunked_data_with_padding_chunk_boundaries(
+    slicing_dim: Literal[0, 1, 2], boundary: Literal["before", "after"]
+):
+    block_shape = [10, 10, 10]
+    block_shape[slicing_dim] = 6
+    start_index = -2 if boundary == "before" else 6
+    data = np.ones(block_shape, dtype=np.float32)
+    global_index = [0, 0, 0]
+    global_index[slicing_dim] = 10 + start_index
+    global_shape_t = [10, 10, 10]
+    global_shape_t[slicing_dim] = 30
+    global_shape = make_3d_shape_from_shape(global_shape_t)
+    chunk_shape_t = [10, 10, 10]
+    chunk_shape_t[slicing_dim] += 4  # for padding
+    chunk_shape = make_3d_shape_from_shape(chunk_shape_t)
+    angles = np.linspace(0, math.pi, global_shape[0], dtype=np.float32)
+    block = DataSetBlock(
+        data=data,
+        aux_data=AuxiliaryData(angles=angles),
+        slicing_dim=slicing_dim,
+        block_start=start_index,
+        chunk_start=10,
+        global_shape=global_shape,
+        chunk_shape=chunk_shape,
+        padding=(2, 2),
+    )
+
+    assert block.is_padded is True
+    assert block.padding == (2, 2)
+    assert block.is_last_in_chunk is (boundary == "after")
+
+
+@pytest.mark.parametrize("slicing_dim", [0, 1, 2])
+@pytest.mark.parametrize("boundary", ["before", "after"])
+def test_partial_block_with_padding_global_boundaries(
+    slicing_dim: Literal[0, 1, 2], boundary: Literal["before", "after"]
+):
+    block_shape = [10, 10, 10]
+    block_shape[slicing_dim] = 6
+    padding = (2, 2)
+    start_index = -padding[0] if boundary == "before" else 6
+    data = np.ones(block_shape, dtype=np.float32)
+    chunk_shape_t = [10, 10, 10]
+    chunk_shape_t[slicing_dim] += padding[0] + padding[1]  # for padding
+    chunk_shape = make_3d_shape_from_shape(chunk_shape_t)
+    global_index = [0, 0, 0]
+    global_index[slicing_dim] = (
+        -padding[0]
+        if boundary == "before"
+        else 30 - block_shape[slicing_dim] + padding[1]
+    )
+    chunk_start = -padding[0] if boundary == "before" else 20 - padding[0]
+    global_shape_t = [10, 10, 10]
+    global_shape_t[slicing_dim] = 30
+    global_shape = make_3d_shape_from_shape(global_shape_t)
+    angles = np.linspace(0, math.pi, global_shape[0], dtype=np.float32)
+    block = DataSetBlock(
+        data=data,
+        aux_data=AuxiliaryData(angles=angles),
+        slicing_dim=slicing_dim,
+        block_start=start_index,
+        chunk_start=chunk_start,
+        global_shape=global_shape,
+        chunk_shape=chunk_shape,
+        padding=padding,
+    )
+
+    assert block.is_padded is True
+    assert block.padding == padding
+    assert block.global_index == tuple(global_index)
+    assert block.is_last_in_chunk is (boundary == "after")
```

Comment:
I see some magic numbers, like 6, 30, and 20, that get repeated in several places: could they be put into vars with names that explain what they are, and then reused where necessary?

Reply:
I reworked all the magic numbers in the tests - please see if it's easier to follow now.